### ISSUE TITLE
New lines break the QR code URI

### ISSUE THREAD

**Issue by taigrr** [open-keychain/open-keychain#2339]
Wiki page: https://github.com/open-keychain/open-keychain/wiki/QR-Codes

This should be an easy fix, I'm just recommending a hint be added to the wiki mentioning that newlines at the end of a OPENPGP4FPR:<FINGERPRINT> URI results in 
```
Fingerprint is malformed or too short!
```
Please add a hint under the expalnation of this URI scheme telling users who generate qrcodes via the cli to strip newlines.

Reproducing:

I first tried to create a QR code with my own key like this:
(Arch Linux, gpg):
Import my key
`gpg --recv 0x616501B35FEE6151`

create a base64 encoded image of my key:
```
gpg --fingerprint Tai Groot |grep 615 |tail -c 51 |tr -d ' '|awk '{print "OPENPGP4FPR:"$1}'|qrencode -o - |base64
```

Once I realized trailing newlines weren't allowed:

```
gpg --fingerprint Tai Groot |grep 615 |tail -c 51 |tr -d ' '|awk '{print "OPENPGP4FPR:"$1}'|tr -d '\n'|qrencode -o - |base64
```
results in the following:

![image](https://user-images.githubusercontent.com/8261498/40823275-a9432796-6524-11e8-9aee-7034aff55da5.png)
Which works correctly!

**Comment by wiktor-k** [comment]
Yes, I've also seen it and spend too much time debugging why a QR doesn't work while it decoded correctly. I even thought about relaxing the scanner to allow trailing whitespace. @Valodim what do you think about this issue?

**Comment by taigrr** [comment]
Edited to make the one-liners readable, added escaped line-wraps.

**Comment by taigrr** [comment]
Hey, just bumping this issue. Should I attempt to fix it and submit a pull request?

**Comment by Valodim** [comment]
Sure, please do :)

**Comment by wiktor-k** [comment]
Just out of curiosity what would the fix do? Allow newlines in QR codes? (that I think should be documented as this won't be "pure" openpgp4fpr URI). Or maybe just display a more informative message or wanting (so that the user can fix their generation script)?

**Comment by taigrr** [comment]
I'm a little torn on that. In one sense I don't think that it should deviate from the spec, but also I think it's a little bit dumb to make the user regenerate everything if all that's different is a newline. 

Maybe interpret it, but give a warning message in the app itself, and also add a note in the documentation. 

I don't like the idea of having an app say "I know what you meant, but too bad, do it right!" But that might just be me and my Arch principles, "functionality over ideology" etc.
